.github/workflows/on-merge-branch-changeset-release-main.yml

```diff
@@ -8,6 +8,13 @@ concurrency:
   group: ${{ github.head_ref || github.ref }}
   cancel-in-progress: true
 
+permissions:
+  contents: read
+
+defaults:
+  run:
+    shell: bash -euo pipefail {0}
```

Comment:
- `-e`: Exits when there's an error.
- `-u`: Fail if a variable is unassigned.
- `-o pipefail`: Fail if any part of a pipe fails.